src/main/java/it/gov/pagopa/pu/workflow/service/WorkflowServiceImpl.java

```diff
@@ -0,0 +1,61 @@
+package it.gov.pagopa.pu.workflow.service;
+
+import com.uber.m3.tally.NoopScope;
+import io.temporal.api.common.v1.WorkflowExecution;
+import io.temporal.api.workflow.v1.WorkflowExecutionInfo;
+import io.temporal.client.WorkflowClient;
+import io.temporal.client.WorkflowOptions;
+import io.temporal.failure.TemporalException;
+import io.temporal.internal.client.WorkflowClientHelper;
+import it.gov.pagopa.pu.workflow.dto.generated.WorkflowStatusDTO;
+import it.gov.pagopa.pu.workflow.exception.custom.WorkflowInternalErrorException;
+import it.gov.pagopa.pu.workflow.exception.custom.WorkflowNotFoundException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class WorkflowServiceImpl implements WorkflowService {
+
+  private final WorkflowClient workflowClient;
+
+  public WorkflowServiceImpl(WorkflowClient workflowClient) {
+    this.workflowClient = workflowClient;
+  }
+
+  @Override
+  public <T> T buildWorkflowStub(Class<T> workflowClass, String taskQueue, String workflowId) {
+    return workflowClient.newWorkflowStub(
+      workflowClass,
+      WorkflowOptions.newBuilder()
+        .setTaskQueue(taskQueue)
+        .setWorkflowId(workflowId)
+        .build());
+  }
+
+  @Override
+  public WorkflowStatusDTO getWorkflowStatus(String workflowId) {
+    try {
+      log.info("Retrieving workflow status for workflowId: {}", workflowId);
```

Comment:
I would prefer a INFO log always on the Controller, putting it also here would let to a duplicate log, let's change it into DEBUG level ;)
```suggestion
      log.debug("Retrieving workflow status for workflowId: {}", workflowId);
```